src/core/lib/surface/init.cc

```diff
@@ -216,3 +218,20 @@ void grpc_maybe_wait_for_async_shutdown(void) {
     g_shutting_down_cv->Wait(g_init_mu);
   }
 }
+
+bool grpc_wait_for_shutdown_with_timeout(absl::Duration timeout) {
+  GRPC_TRACE_LOG(api, INFO) << "grpc_wait_for_shutdown_with_timeout()";
+  const auto started = absl::Now();
+  const auto deadline = started + timeout;
+  gpr_once_init(&g_basic_init, do_basic_init);
+  grpc_core::MutexLock lock(g_init_mu);
+  while (g_initializations != 0) {
+    if (g_shutting_down_cv->WaitWithDeadline(g_init_mu, deadline)) {
+      LOG(ERROR) << "grpc_wait_for_shutdown_with_timeout() timed out.";
+      return false;
+    }
+  }
+  LOG(INFO) << "grpc_wait_for_shutdown_with_timeout() took "
```

Comment:
Do we really want `LOG(INFO)` here? I think `VLOG(2)` may be more appropriate, comparing to the rest of this file.

Reply:
Changed to trace only. I think that gives a finer control.